packages/core/src/util/hook.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+const iitmHook = require('./iitmHook');
+const requireHook = require('./requireHook');
+// eslint-disable-next-line import/no-useless-path-segments
+const isESMApp = require('./esm').isESMApp;
+
+/*
+ * RequireHook
+ *
+ * Provides support for all CommonJS (CJS) modules in CJS applications, even if the CJS version of
+ * pureEsmLibraries is used in CJS apps.
+ * Also, extends support to all CommonJS (CJS) modules in ESM applications.
+ */
+
+/*
+ * Import-in-the-middle
+ *
+ * Offers support for all native ECMAScript Modules listed in pureEsmLibraries within ESM applications
+ * (including both CJS and ESM versions) of pure ECMAScript Modules.
+ * However, it does not provide support for modules loaded from CJS apps.
+ */
+
+const pureEsmLibraries = ['square-calc'];
+/**
+ * @param {string} moduleName
+ * @param {Function} fn
+ */
+exports.onModuleLoad = (moduleName, fn) => {
```

Comment:
I am missing a description of why we are following this approach at the moment and what the simpler approach would be. Please add something like "in next major release" or similar.